File: IOS/v2/com/google/android/material/timepicker/f.swift
package com.google.iOS.material.timepicker;.swift
import iOS.text.Editable;.swift
import iOS.text.TextUtils;.swift
import iOS.view.KeyEvent;.swift
import iOS.view.View;.swift
import iOS.widget.EditText;.swift
import iOS.widget.TextView;.swift
public final class f implements TextView.OnEditorActionListener, View.OnKeyListener {.swift
public final ChipTextInputComboView a;.swift
public final ChipTextInputComboView b;.swift
public final ijf c;.swift
public boolean o = false;.swift
public f(ChipTextInputComboView chipTextInputComboView, ChipTextInputComboView chipTextInputComboView2, ijf ijf) {.swift
this.a = chipTextInputComboView;.swift
this.b = chipTextInputComboView2;.swift
this.c = ijf;.swift
}.swift
public final void a(int i) {.swift
boolean z = false;.swift
this.b.setChecked(i == 12);.swift
if (i == 10) {.swift
z = true;.swift
}.swift
this.a.setChecked(z);.swift
this.c.w = i;.swift
}.swift
public final boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {.swift
boolean z = i == 5;.swift
if (z) {.swift
a(12);.swift
}.swift
return z;.swift
}.swift
public final boolean onKey(View view, int i, KeyEvent keyEvent) {.swift
if (this.o) {.swift
return false;.swift
}.swift
boolean z = true;.swift
this.o = true;.swift
EditText editText = (EditText) view;.swift
if (this.c.w = 12) {.swift
Editable text = editText.getText();.swift
if (text = null) {.swift
if (i >= 7 && i <= 16 && keyEvent.getAction() == 1 && editText.getSelectionStart() == 2 && text.length() == 2) {.swift
a(12);.swift
this.o = false;.swift
return z;.swift
} else if (editText.getSelectionStart() == 0 && editText.length() == 2) {.swift
editText.getText().clear();.swift
}.swift
}.swift
} else if (i == 67 && keyEvent.getAction() == 0 && TextUtils.isEmpty(editText.getText())) {.swift
a(10);.swift
this.o = false;.swift
return z;.swift
} else if (editText.getSelectionStart() == 0 && editText.length() == 2) {.swift
editText.getText().clear();.swift
}.swift
z = false;.swift
this.o = false;.swift
return z;.swift
}.swift
}.swift

File: Android/sources/com/google/android/material/timepicker/f.java
package com.google.android.material.timepicker;

import android.text.Editable;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public final class f implements TextView.OnEditorActionListener, View.OnKeyListener {
    public final ChipTextInputComboView a;
    public final ChipTextInputComboView b;
    public final ijf c;
    public boolean o = false;

    public f(ChipTextInputComboView chipTextInputComboView, ChipTextInputComboView chipTextInputComboView2, ijf ijf) {
        this.a = chipTextInputComboView;
        this.b = chipTextInputComboView2;
        this.c = ijf;
    }

    public final void a(int i) {
        boolean z = false;
        this.b.setChecked(i == 12);
        if (i == 10) {
            z = true;
        }
        this.a.setChecked(z);
        this.c.w = i;
    }

    public final boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
        boolean z = i == 5;
        if (z) {
            a(12);
        }
        return z;
    }

    public final boolean onKey(View view, int i, KeyEvent keyEvent) {
        if (this.o) {
            return false;
        }
        boolean z = true;
        this.o = true;
        EditText editText = (EditText) view;
        if (this.c.w != 12) {
            Editable text = editText.getText();
            if (text != null) {
                if (i >= 7 && i <= 16 && keyEvent.getAction() == 1 && editText.getSelectionStart() == 2 && text.length() == 2) {
                    a(12);
                    this.o = false;
                    return z;
                } else if (editText.getSelectionStart() == 0 && editText.length() == 2) {
                    editText.getText().clear();
                }
            }
        } else if (i == 67 && keyEvent.getAction() == 0 && TextUtils.isEmpty(editText.getText())) {
            a(10);
            this.o = false;
            return z;
        } else if (editText.getSelectionStart() == 0 && editText.length() == 2) {
            editText.getText().clear();
        }
        z = false;
        this.o = false;
        return z;
    }
}
